frontend/src/components/browse-sightings-page/BrowseSightings.tsx

```diff
@@ -1,25 +1,43 @@
 import React, { useEffect, useState } from "react";
-import { getSightings, Sighting } from "../../clients/apiClient";
+import {
+  getSightings,
+  getSightingsBySpeciesId,
+  SightingWithLocations,
+} from "../../clients/apiClient";
 import { SightingCard } from "./SightingCard";
 import "./BrowseSightings.scss";
+import { useParams } from "react-router-dom";
 
 export const BrowseSightings: React.FC = () => {
-  const [sightings, setSightings] = useState<Sighting[]>();
+  const [sightings, setSightings] = useState<SightingWithLocations[]>();
+  const { speciesId } = useParams<{ speciesId: string }>();
 
-  useEffect(() => {
-    getSightings().then(setSightings);
-  }, []);
+  if (speciesId === undefined) {
+    useEffect(() => {
+      getSightings().then(setSightings);
+    }, []);
+  } else {
+    useEffect(() => {
+      getSightingsBySpeciesId(speciesId).then(setSightings);
+    }, [speciesId]);
+  }
+
+  let pageHeadline = "";
 
   if (sightings === undefined) {
-    return <p>Loading</p>;
+    pageHeadline = "Loading";
+  } else if (sightings.length === 0) {
+    pageHeadline = "Sorry, no whales of that species have been seen";
+  } else {
+    pageHeadline = "Reported Sightings";
   }
 
   return (
     <>
-      <h1>Reported Sightings</h1>
+      <h1>{pageHeadline}</h1>
       <ul>
-        {sightings.map((sighting, index) => (
-          <SightingCard sighting={sighting} key={sighting.id} />
+        {sightings?.map((sighting, index) => (
+          <SightingCard sighting={sighting} key={index} />
```

Comment:
`key` shouldn't be `index`, use `sighting.id` instead
```suggestion
        {sightings?.map((sighting) => (
          <SightingCard sighting={sighting} key={sighting.id} />
```